lua/starfall/editor/sfframe.lua

```diff
@@ -1663,6 +1703,117 @@ function Editor:LoadFile(Line, forcenewtab)
 	end
 end
 
+---Returns the value of the settings `ReloadBeforeUpload` of the editor.
+---@return boolean
+function Editor:ShouldReloadBeforeUpload()
+    return self.autoReloadEnabled
+end
+
+---Reloads the tab associated to the file at `filepath`, if there is one.
+---@param tabIndex number The index of the tab to reload
+---@param interactive boolean If the file has unsaved changed and interactive is true
+---then prompt the user to overwrite the current unsaved changes, otherwise dont reload the file.
+function Editor:ReloadTab(tabIndex, interactive)
+	local activeTabIndex = self:GetActiveTabIndex()
+	local tab = self:GetTab(tabIndex)
+	local tabContent = self:GetTabContent(tabIndex)
+	if not tabContent:GetTabHandler().IsEditor then return end
+
+	local filepath = tabContent.chosenfile
+	if filepath == nil then
+		-- Some tabs can have this field set to nil
+		-- At least the default "Generic" tab seems to have
+		return
+	end
+
+	local fileContent = file.Read(filepath)
+	if fileContent == nil then
+		ErrorNoHalt("Error while reloading, failed to read file: ", filepath)
+		return
+	end
+	local fileLastModified = file.Time(filepath, "DATA")
```

Comment:
I think the file.Time should be checked before file.Read for performance reasons.

Reply:
Should be fixed now.